packages/core/admin/admin/src/components/LanguageProvider/init.js

```diff
@@ -0,0 +1,15 @@
+const localStorageKey = 'strapi-admin-language';
+
+const init = localesNativeNames => {
+  const languageFromLocaleStorage = window.localStorage.getItem(localStorageKey);
+  const appLanguage = localesNativeNames[languageFromLocaleStorage]
+    ? languageFromLocaleStorage
+    : 'en';
```

Comment:
Same here about the default browser locale. Shouldn't we choose the default browser language and set English if the default browser language doesn't exist in our list like it was done before? IMO it could be a quick win

Reply:
I'd prefer sticking to the supported locales instead.

Reply:
You can check if the browser locale is supported. If not, we can choose English by default?